dockerfiles/hack/run-integration.py

```diff
@@ -55,8 +56,13 @@ def build_args():
         args.extend(INTEGRATION_EXTRA_ARGS.split())
     return args
 
+def refresh_vault():
+    while True:
+        VaultClient.refresh_client()
+        time.sleep(60*10)
 
 if __name__ == "__main__":
+    threading.Thread(target=refresh_vault, daemon=True).start()
```

Comment:
This should be a feature of the VaultClient, imo. Callers should not have to worry about that.